packages/delegation-items-archiver/src/delegationItemsArchiverProcessors.ts

```diff
@@ -0,0 +1,74 @@
+import { match } from "ts-pattern";
+import { DelegationId, purposeVersionState } from "pagopa-interop-models";
+import { InteropHeaders } from "pagopa-interop-commons";
+import { PurposeProcessClient } from "./clients/clientsProvider.js";
+import { ReadModelService } from "./readModelService.js";
+
+export const processPurposes = async ({
+  readModelService,
+  purposeProcessClient,
+  headers,
+  delegationId,
+}: {
+  readModelService: ReadModelService;
+  purposeProcessClient: PurposeProcessClient;
+  headers: InteropHeaders;
+  delegationId: DelegationId;
+}): Promise<void> => {
+  const purposes = await readModelService.getPurposes(delegationId);
+
+  await Promise.all(
+    purposes.map(async (p) => {
+      const isDeletable = p.versions.every((v) =>
+        match(v.state)
+          .with(
+            purposeVersionState.draft,
+            purposeVersionState.waitingForApproval,
+            () => true
+          )
+          .with(
+            purposeVersionState.archived,
+            purposeVersionState.rejected,
+            purposeVersionState.active,
+            purposeVersionState.suspended,
+            () => false
+          )
+          .exhaustive()
+      );
+      if (isDeletable) {
+        return purposeProcessClient.deletePurpose(undefined, {
+          params: { id: p.id },
+          headers,
+        });
+      }
+
+      const activeOrSuspendedVersion = p.versions.find((v) =>
+        match(v.state)
+          .with(
+            purposeVersionState.active,
+            purposeVersionState.suspended,
+            () => true
+          )
+          .with(
+            purposeVersionState.archived,
+            purposeVersionState.rejected,
+            purposeVersionState.draft,
+            purposeVersionState.waitingForApproval,
+            () => false
+          )
+          .exhaustive()
+      );
+      if (activeOrSuspendedVersion) {
+        return purposeProcessClient.archivePurposeVersion(undefined, {
```

Comment:
I suppose there is no way to archive the entire purpose?
In this way we keep the purpose with the `delegationId` and just archive all its versions while the SRS says to archive the purpose

Reply:
👌 anyway the purpose itself does not have a state, so maybe the new dedicated route could just archive all the purpose versions... but then it's the same as doing it like this, not strong opinion